### PR TITLE
🗃️ Feat #5 Setup Docker Compose

### PR DESCRIPTION
**Pull request by demariadaniel**
- Add Dockerfile for creating Postgres DB
- Create base migration files
- Correct enum implementation
- Add basic unit tests for insert & delete
- Minor readme updates

### CODE REVIEW

File: apps/api/package.json [hunk 1]
```diff
@@ -5,9 +5,8 @@
 	"type": "module",
 	"main": "src/main.ts",
 	"scripts": {
-		"db-test": "npx tsx --env-file=.env ./src/db/index.ts",
 		"dev": "npx tsx watch --env-file=.env ./src/main.ts",
-		"test": "npx tsx --test --test-reporter=spec --experimental-test-coverage ./src/tests/*"
+		"test": "npx tsx --env-file=.env --test --test-reporter=spec --experimental-test-coverage ./src/tests/**/*"
```
Comment (demariadaniel) [inline apps/api/package.json:9]: We could potentially set up testcontainers now and replace `.env` here with `.env.test`

File: apps/api/src/db/schemas/actions.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ import { revisionRequests } from './revisionRequests.ts';
 // TODO: Integrate w/ TS
 // import { applicationActions } from 'pcgl-daco/packages/data-model/';
 
-const actionsEnum = pgEnum('actions', [
+export const actionTypesEnum = pgEnum('action_types', [
```
Comment (demariadaniel) [inline apps/api/src/db/schemas/actions.ts:27]: Exporting enums due to bug described here: https://github.com/drizzle-team/drizzle-orm/issues/646#issuecomment-1586349095

Also updated to use lower snake case naming conventions; add `types` to distinguish btwn enums + tables

File: README.md [hunk 3]
```diff
@@ -45,9 +45,11 @@ The modules in the monorepo are organized into two categories:
 - Start server:
   
   `cd apps/api`
-  
-  `cp .env.schema .env`
 
+  Create .env file: `cp .env.schema .env`
+
+  Start Postgres DB: `docker compose -f ./docker-compose.yml up --detach`
```
Comment (justincorrigible) [inline README.md:51]: iirc this part shouldn't be necessary
```suggestion
  Start Postgres DB: `docker compose up --detach`
```

File: apps/api/src/db/schemas/applications.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ import { applicationContents } from './applicationContents.ts';
 // TODO: Integrate w/ TS
 // import { ApplicationStates } from 'pcgl-daco/packages/data-model';
 
-const statesEnum = pgEnum('applicationStates', [
+export const statesEnum = pgEnum('application_states', [
```
Comment (justincorrigible) [inline apps/api/src/db/schemas/applications.ts:26]: gotta keep consistent with agreements etc
so we have to either add application to names in this file, or remove agreements there (which looks less descriptive)

better being extra verbose overall, than not enough
```suggestion
export const applicationStatesEnum = pgEnum('application_states', [
```

**Review by justincorrigible — COMMENTED** [review]
nice setup!! some suggestions to make it better

File: apps/api/README.md [hunk 5]
```diff
@@ -24,4 +24,4 @@ This app uses features from Node ^20.9, in combination with TSX for TypeScript s
 
 ### Database
 
-- This application implements a Postgres database using the following [data model](../../docs/model/README.md)
\ No newline at end of file
+- This application implements a Postgres database based on the following [data model](../../docs/model/README.md) and managed by [Drizzle ORM](https://orm.drizzle.team/docs/overview)
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply apps/api/README.md:27]: Sorry easy thing to miss!

[the remark this answers is not in the document]
Comment (justincorrigible) [reply apps/api/README.md:27]: you're right! happens all the time 👍

**Review by justincorrigible — APPROVED** [review]

**Pull request merged**